packages/app/src/components/drawer/networks-drawer.tsx

```diff
@@ -1,72 +1,84 @@
-import React from 'react';
-import { Box, Flex, Text, Button } from '@stacks/ui';
+import React, { memo, useCallback } from 'react';
+import { useSetRecoilState } from 'recoil';
+import { Box, Flex, Button, Stack, color, FlexProps, BoxProps } from '@stacks/ui';
 import { ControlledDrawer } from './controlled';
 import { useWallet } from '@common/hooks/use-wallet';
 import { CheckmarkIcon } from '@components/icons/checkmark-icon';
 import { useAnalytics } from '@common/hooks/use-analytics';
 import { ScreenPaths } from '@store/onboarding/types';
-import { useSetRecoilState } from 'recoil';
 import { currentNetworkKeyStore } from '@store/recoil/networks';
 import { showNetworksStore } from '@store/recoil/drawers';
 import { useDrawers } from '@common/hooks/use-drawers';
+import { Caption, Title } from '@components/typography';
 
-export const NetworksDrawer: React.FC = () => {
+const NetworkListItem: React.FC<{ item: string } & BoxProps> = memo(({ item, ...props }) => {
   const { setShowNetworks } = useDrawers();
-
   const { networks, currentNetworkKey } = useWallet();
   const setCurrentNetworkKey = useSetRecoilState(currentNetworkKeyStore);
-  const { doChangeScreen } = useAnalytics();
-  const networkRows = Object.keys(networks).map(networkKey => {
-    const network = networks[networkKey];
-    return (
-      <Box
-        width="100%"
-        key={networkKey}
-        _hover={{
-          backgroundColor: 'ink.150',
-        }}
-        cursor="pointer"
-        px={6}
-        py="base"
-        onClick={() => {
-          setCurrentNetworkKey(networkKey);
-          setShowNetworks(false);
-        }}
-      >
-        <Flex width="100%">
-          <Box flexGrow={1}>
-            <Text fontSize={2} display="block">
-              {network.name}
-            </Text>
-            <Text fontSize={1} color="gray">
-              {network.url}
-            </Text>
-          </Box>
-          <Box pt="base-loose">{networkKey === currentNetworkKey ? <CheckmarkIcon /> : null}</Box>
-        </Flex>
-      </Box>
-    );
-  });
+
+  const network = networks[item];
+
+  const handleItemClick = useCallback(() => {
+    setCurrentNetworkKey(item);
+    setTimeout(() => {
+      setShowNetworks(false);
+    }, 350); // timeout to show the check mark move :)
```

Comment:
```suggestion
    const delayToShowCheckmarkMove = 350;
    setTimeout(() => {
      setShowNetworks(false);
    }, delayToShowCheckmarkMove);
```
Not needed, but my pref's for named const to explain rather than comment